public/app/plugins/datasource/grafana/components/TimeRegionEditor.tsx

```diff
@@ -0,0 +1,173 @@
+
+import { TimeRegionConfig } from '../types';
+
+import { TimePickerInput } from './TimePickerInput';
+
+interface Props {
+  value: TimeRegionConfig;
+  onChange: (value?: TimeRegionConfig) => void;
+}
+
+const days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'].map((v, idx) => {
+  return {
+    label: v,
+    value: idx + 1,
+  };
+});
+export const TimeRegionEditor = ({ value, onChange }: Props) => {
+  const styles = useStyles2(getStyles);
+  const timestamp = Date.now();
+  const timezoneInfo = getTimeZoneInfo(value.timezone!, timestamp);
+  const isDashboardTimezone = getDashboardSrv().getCurrent()?.getTimezone() === value.timezone;
+
+  const [isEditing, setEditing] = useState(false);
+
+  const onToggleChangeTimezone = () => {
+    setEditing(!isEditing);
+  };
+
+  const getTime = (time: string | undefined): Moment | undefined => {
+    if (!time) {
+      return undefined;
+    }
+
+    const date = moment();
+
+    if (time) {
+      const match = time.split(':');
+      date.set('hour', parseInt(match[0], 10));
+      date.set('minute', parseInt(match[1], 10));
+    }
+
+    return date;
+  };
+
+  const getToPlaceholder = () => {
+    let placeholder = 'Everyday';
+    if (value.fromDayOfWeek && !value.toDayOfWeek) {
+      placeholder = days[value.fromDayOfWeek - 1].label;
+    }
+
+    return placeholder;
+  };
+
+  const renderTimezonePicker = () => {
+    const timezone = (
+      <>
+        <TimeZoneTitle title={timezoneInfo?.name} />
+        <TimeZoneOffset timeZone={value.timezone} timestamp={timestamp} />
+      </>
+    );
+
+    if (isDashboardTimezone) {
+      return <>Dashboard timezone ({timezone})</>;
+    }
+
+    return timezone;
+  };
+
+  const renderTimezone = () => {
+    if (isEditing) {
+      return (
+        <TimeZonePicker
+          value={value.timezone}
+          includeInternal={true}
+          onChange={(v) => onChange({ ...value, timezone: v })}
+          onBlur={() => setEditing(false)}
+          menuShouldPortal={true}
+          openMenuOnFocus={false}
+          width={100}
+          autoFocus
+        />
+      );
+    }
+
+    return (
+      <div className={styles.timezoneContainer}>
+        <div className={styles.timezone}>{renderTimezonePicker()}</div>
+        <Button variant="secondary" onClick={onToggleChangeTimezone} size="sm">
+          Change timezone
+        </Button>
+      </div>
+    );
+  };
+
+  return (
+    <FieldSet className={styles.wrapper}>
+      <Field label="From">
+        <HorizontalGroup spacing="xs">
+          <Select
+            options={days}
+            isClearable
+            placeholder="Everyday"
+            value={value.fromDayOfWeek ?? null}
+            onChange={(v) => {
+              const fromDayOfWeek = v ? v.value : undefined;
+              const toDayOfWeek = v ? value.toDayOfWeek : undefined; // clear if everyday
+              onChange({ ...value, fromDayOfWeek, toDayOfWeek });
+            }}
```

Comment:
may be a bit cleaner to break out the inline onChange funcs below into stand alone functions defined above the return on 104